qa/restart-upgrade/build.gradle

```diff
@@ -102,6 +102,7 @@ task testAgainstOldCluster(type: StandaloneRestIntegTestTask) {
     if (versionsBelow2_15.any { ext.neural_search_bwc_version.startsWith(it) }){
         filter {
             excludeTestsMatching "org.opensearch.neuralsearch.bwc.NeuralSparseTwoPhaseProcessorIT.*"
+            excludeTestsMatching "org.opensearch.neuralsearch.bwc.HybridSearchWithRescoreIT.*"
```

Comment:
this check is not needed in main, latest 2.x which is 2.18 already supports rescore.

Reply:
I prefer to keep it so that the code looks same between `main` and `2.x`